sktime/forecasting/model_selection/_split.py

```diff
@@ -521,6 +522,17 @@ def get_fh(self) -> ForecastingHorizon:
     def __repr__(self) -> str:
         return _repr(self)
 
+    @staticmethod
+    def _get_train_window(
+        y: ACCEPTED_Y_TYPES, train_start: int, split_point: int
+    ) -> np.ndarray:
+        if (split_point <= len(y)) & (split_point > 0):
+            return np.argwhere(
+                (y >= y[max(train_start, 0)]) & (y <= y[min(split_point, len(y)) - 1])
+            ).flatten()
+        else:
+            return np.array([], dtype=np.int)
```

Comment:
is this correct? This does not look right:

* should it not be empty only if the split point is in the past of the `train_start`, otherwise non-empty?
* it also seems this is correct only for an index? Namely, if `y` is not an index, you'll slice to the series where the *values* are between the starting and end values, not the *index* being between the start/end index!
* this also looks like sth that could 1:1 be replaced by `get_window`, if applied to index
* given that it seems incorrect and identical to `get_window`, consider replacing?

Reply:
(depends on the specification of course, so I may be wrong - docstrings would be appreciated)

Reply:
- changed to
```
if split_point > max(0, train_start):
```
- added docstring. `y` is always an index here due to uniform usage of `_check_y` function across all splitters.
- `get_window` is not applicable here. This method takes start/end indices, not window length.

Reply:
> This method takes start/end indices, not window length.

end = start + length?

The real "burden" is with treating the multiple possible data format types, not in the selection, is it?

Reply:
Look, I will not be using this function. At least not in this PR. It operates with window distances, while `_get_train_window` operates with exact points in the index. Distances are computed outside. Just leave it, please.

Reply:
ok, sure